lasso/transit.py

```diff
@@ -298,18 +298,33 @@ def evaluate_differences(self, base_transit):
         """
         Evaluate Additions
 
-        First assess if need to add multiple routes if there are multiple time periods
         """
-        for line in lines_to_add:
-            time_period_numbers = CubeTransit.get_time_period_numbers_from_cube_properties(
-                self.line_properties[line]
-            )
-            if len(time_period_numbers) > 1:
-                for tp in time_period_numbers[1:]:
-                    lines_to_add.append(self.add_additional_time_periods(tp, line))
+        added_routes=[]
+        add_card_dict = {
+            "category": "Add New Route",
+            "routes": []
+            }
 
         for line in lines_to_add:
-            add_card_dict = self.create_add_route_card_dict(line)
+
+            route_properties, trip_properties = self.create_routing_properties(line)
+
+            # check these attributs to see if a route is an existing added route
+            # if yes, then nest the trips to existing route
+            route_match_attributes = ["route_id", "route_short_name", "route_long_name", 
+                                "route_type", "agency_raw_name", "agency_id"]
+
+            if route_properties in added_routes:
+                for route in add_card_dict['routes']:
+                    if all(route[attr] == route_properties[attr] for attr in route_match_attributes):
+                        route['trips'].append(trip_properties)
+
+            else:
+                added_routes.append(route_properties.copy())
+                route_properties['trips'] = [trip_properties]
+                add_card_dict["routes"].append(route_properties)
+
+        if len(added_routes)>0:
```

Comment:
What is the reason for adding this filter? What is the case for not appending `add_card_dict` when `lines_to_add` is not empty? Please add comments to this filter.